app/models/message.rb

```diff
@@ -168,7 +169,8 @@ def update_contact_activity
   def dispatch_create_events
     Rails.configuration.dispatcher.dispatch(MESSAGE_CREATED, Time.zone.now, message: self, performed_by: Current.executed_by)
 
-    if outgoing? && conversation.messages.outgoing.count == 1
+    campaign_outgoing_messages = conversation.messages.outgoing.where(created_by: 'Campaign')
```

Comment:
I would suggest saving campaign_id instead of "campaign" text, so it is easy for us to identify the campaign from which the message was triggered.

Reply:
Saved `campaign_id`  in `additional_attributes`.